### PR TITLE
[INVE-12911] Support Hapi 20

### PR DESCRIPTION
**Pull request by daradermody**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -54,17 +54,16 @@ internals.schema = Joi.object({
...
-    server.handler('kibi_proxy', internals.handler);
+    server.decorate('handler', 'kibi_proxy', internals.handler);
 
-    server.decorate('reply', 'kibi_proxy', function (options) {
+    server.decorate('toolkit', 'kibi_proxy', function (options) {
 
         internals.handler(this.request.route, options)(this.request, this);
     });
-
-    return next();
```
Comment (szydan) [inline lib/index.js:67]: No need to return next() anyore?

**Review by szydan — CHANGES_REQUESTED** [review]
Can you paste the output of the tests please 
https://github.com/sirensolutions/h2o2/blob/master/package.json#L31

File: lib/index.js (hunk 1 above)
Comment (daradermody) [reply lib/index.js:67]: All the methods that used to require callbacks (e.g. `next()`) are now asynchronous. Return a promise and it is `await`ed instead.
image

**Comment by daradermody** [comment]
> Can you paste the output of the tests please

@szydan I didn't update the tests because they seemed to already be failing, so I figured no one cared about them anyway (since they're not even run as part of the PR). If you think they are important enough, I can spend the time fixing the tests.

**Comment by szydan** [comment]
@daradermody please lets try.

**Comment by daradermody** [comment]
> @daradermody please lets try.

@szydan Is it fine to stop at 13 fixed tests? Doing all 60+ will take several days and might be better suited in a separate technical debt ticket.

image

**Comment by szydan** [comment]
@daradermody  I've cloned the h2o2 version 9.0.2 
```
~/home/workspace/h2o2 on (no branch)△
$ npm test 

> @hapi/h2o2@9.0.2 test /Users/szydan/home/workspace/h2o2
> lab -a @hapi/code -t 100 -L


  
  ..........................(node:54741) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
........................
  .................

67 tests complete
Test duration: 682 ms
Assertions count: 147 (verbosity: 2.19)
Leaks: No issues
Coverage: 100.00%
Lint: No issues
```

Can we instead of fixing the old code start from this version of h2o2 9.02 
and port our lines to it 
Also try to port our tests - they should pass 
Then run the tests and see how many will fail 
Then maybe try to fix them or if too many would fail we open a new ticket 
This module is too "core" (all proxies are using it)
I am scared to use our version without having tests 
It is sad that the tests were not maintained by as before ;-( but lets try now 
if it takes time it is fine , we knew that bumping happy will not be easy

**Comment by daradermody** [comment]
Closing in favour of https://github.com/sirensolutions/h2o2/pull/13, which pulls all the latest code from [hapijs/h2o2](https://github.com/hapijs/h2o2) and simplifies the fork changes.

**Pull request closed without merging**